### ISSUE TITLE
Issues with v6.0.4

### ISSUE THREAD

**Issue by bart-degreed** [sethreno/ReGitLint#14]
Hi @sethreno,

Thanks for addressing open bugs. I'd like to upgrade to the latest version, but have some concerns.
- You're targeting .NET 6 now. I suspect that now it has to be installed on build servers to use it, but we're not there yet. This is a huge breaking change, as the previous minor version worked on .NET 3.1 and .NET 5.
- The fix for #10 now matches too much in all cases. Our VMs on build server are slow already, this makes things worse for everyone. It would be nice to put the wider matching behind a command-line switch, so only users with their .sln in a subfolder can activate this.

**Comment by sethreno** [comment]
Hey @bart-degreed, Thanks for reporting. I'm happy to try and address these issues, but might need a bit of guidance on the first:

* I'm targeting .net 6 because it's the latest LTS release and I didn't want to install older runtimes in my latest images. Do you know if there is way to publish a dotnet tool that targets both or do you have any other suggestions?

* Adding an opt-in switch for this hack is probably the right way to go. I considered it, but initially decided against it for the sake of remaining "low friction". On further consideration though I'm guessing the majority of users have the .sln file in their root?

**Comment by bart-degreed** [comment]
I've never created a global tool, but I suspect it's not much different than [NuGet multi-targeting](https://docs.microsoft.com/en-us/nuget/create-packages/supporting-multiple-target-frameworks) in general.

I may be as simple as replacing:
```xml
<TargetFramework>net6.0</TargetFramework>
```
with:
```xml
<TargetFrameworks>netcoreapp3.1;net6.0</TargetFrameworks>
```
in `ReGitLint.csproj`.

Note you'll need to change to TargetFrameworks (plural).

And yes, having the .sln at the root is quite common. It's what Visual Studio does when creating a new project.

**Comment by sethreno** [comment]
Thanks for your help on this. I addressed both issues you mentioned in v6.0.6. I decided to go with an opt-out option instead of opt-in. `--disable-jb-path-hack`

My thought was that it keeps it "low friction" for everyone and the false positive matches seem unlikely because each file still includes the full path. e.g. `**/src/AppA/Main.cs` wont't match `**/src/App2/Main.cs`

**Comment by bart-degreed** [comment]
Thanks for adding the switch. ~However, the NuGet package still only contains a binary for .NET 6, so this problem remains.~ My mistake, I was looking at the wrong version.
